pkg/asset/manifests/azure/cluster.go

```diff
@@ -91,6 +92,21 @@ func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID
 		nodeSubnetID = "UNKNOWN"
 	}
 
+	virtualNetworkID := ""
+	if installConfig.Config.Azure.VirtualNetwork != "" {
+		client, err := installConfig.Azure.Client()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get azure client: %w", err)
+		}
+
+		virtualNetwork, err := client.GetVirtualNetwork(context.TODO(), installConfig.Config.Azure.NetworkResourceGroupName, installConfig.Config.Azure.VirtualNetwork)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get azure client: %w", err)
+		}
+
```

Comment:
Even when no error is returned, maybe a good idea to check if virtualNetwork != nil before accessing the ID.